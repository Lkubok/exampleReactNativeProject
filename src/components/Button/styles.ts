import { StyleSheet } from 'react-native';

import { variables, fontNames } from 'res';

export default StyleSheet.create({
  buttonStyle: {
    borderColor: variables.colors.accentRed,
    borderRadius: variables.borderRadius.base,
    borderWidth: 2,
    height: 60,
  },
  buttonWhiteStyle: {
    borderRadius: variables.borderRadius.small,
    borderWidth: 0,
    height: 56,
  },
  deselectText: {
    color: variables.colors.accentGreen,
  },
  headerWrapper: {
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  iconWrapper: {
    marginRight: variables.spacing.min,
    marginTop: 2,
  },
  indicator: {
    flex: 1,
  },
  scrollContainer: {
    backgroundColor: variables.colors.white,
  },
  scrollList: {
    backgroundColor: variables.colors.white,
  },
  scrollStyle: {
    backgroundColor: variables.colors.white,
  },
  tabsContainer: {
    paddingHorizontal: 24,
    paddingTop: 12,
    width: '100%',
  },
  titleStyle: {
    fontFamily: fontNames.SFCompactTextSemibold,
    fontSize: variables.fontSize.medium,
    letterSpacing: variables.letterSpacing.base,
  },
  titleWhiteStyle: {
    color: variables.colors.steel,
    fontSize: 16,
  },
});
