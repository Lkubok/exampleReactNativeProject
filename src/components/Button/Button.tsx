import React, { FunctionComponent } from 'react';
import { Text, View, ViewStyle, StyleProp, StyleSheet } from 'react-native';
import { Button as RneButton } from 'react-native-elements';
import { variables } from 'res';

import styles from './styles';

interface Props {
  disabled?: boolean;
  icon?: Element;
  title: string | Element;
  type?: string;
  onPress: () => void;
  style?: StyleProp<ViewStyle>;
  loading?: boolean;
}

const Button: FunctionComponent<Props> = ({ disabled, icon, onPress, title, type, style, loading }) => {
  let bgColor = 'transparent';
  let borderColor = variables.colors.accentRed;

  if (type === 'white') {
    bgColor = variables.colors.white;
  }
  if (type === 'red') {
    bgColor = variables.colors.accentRed;
  } else if (type === 'green') {
    bgColor = variables.colors.accentGreen;
    borderColor = variables.colors.accentGreen;
  } else if (type === 'outline') {
    bgColor = '';
  }
  if (disabled) {
    bgColor = variables.colors.pinkishGrey;
    borderColor = variables.colors.pinkishGrey;
  }

  return (
    <RneButton
      buttonStyle={StyleSheet.flatten([
        styles.buttonStyle,
        { backgroundColor: bgColor, borderColor: borderColor },
        type === 'white' && styles.buttonWhiteStyle,
        ,
      ])}
      containerStyle={StyleSheet.flatten([style])}
      disabled={disabled}
      disabledStyle={{
        backgroundColor: type !== 'outline' ? variables.colors.pinkishGrey : variables.colors.transparent,
        borderColor: variables.colors.pinkishGrey,
      }}
      disabledTitleStyle={{
        color: type !== 'outline' ? variables.colors.white : variables.colors.pinkishGrey,
      }}
      titleStyle={StyleSheet.flatten([
        { color: type === 'outline' ? variables.colors.accentRed : variables.colors.white },
        styles.titleStyle,
      ])}
      icon={icon && <View style={styles.iconWrapper}>{icon}</View>}
      title={
        <Text
          style={StyleSheet.flatten([
            { color: type === 'outline' ? variables.colors.accentRed : variables.colors.white },
            styles.titleStyle,
            type === 'white' && styles.titleWhiteStyle,
          ])}>
          {title}
        </Text>
      }
      loading={loading}
      onPress={onPress}
    />
  );
};

export default Button;
