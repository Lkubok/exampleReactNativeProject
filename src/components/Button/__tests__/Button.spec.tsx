import React from 'react';
import { act, customRender } from 'utils/testUtils';

import Button from '../Button';

jest.useFakeTimers();

describe('Button component', () => {
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  let props: any;
  beforeEach(() => {
    props = {
      disabled: true,
      loading: false,
      onPress: () => {},
      title: 'testingButton',
      useNativeDriver: true,
    };
  });

  it('should match snapshot', () => {
    const snapshot = customRender(<Button {...props} />).toJSON();
    // eslint-disable-next-line no-void
    void act(() => {
      expect(snapshot).toMatchSnapshot();
    });
  });
});
