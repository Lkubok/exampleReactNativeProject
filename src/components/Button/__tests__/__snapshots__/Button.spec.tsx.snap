// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should match snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "borderRadius": 30,
        "overflow": "hidden",
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "busy": false,
          "disabled": true,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "[7m[38;2;198;198;198m#c6c6c6[39m[27m",
            "borderColor": "[7m[38;2;198;198;198m#c6c6c6[39m[27m",
            "borderRadius": 30,
            "borderWidth": 2,
            "flexDirection": "row",
            "height": 60,
            "justifyContent": "center",
            "padding": 8,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "[7m[38;2;255;255;255m#ffffff[39m[27m",
              "fontFamily": "SFCompactText-Semibold",
              "fontSize": 18,
              "letterSpacing": 0.29,
              "paddingBottom": 1,
              "paddingTop": 2,
              "textAlign": "center",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "[7m[38;2;255;255;255m#ffffff[39m[27m",
                "fontFamily": "SFCompactText-Semibold",
                "fontSize": 18,
                "letterSpacing": 0.29,
              }
            }
          >
            testingButton
          </Text>
        </Text>
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
